### PR TITLE
Materialize progress bar

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: packages/video_player/lib/video_player.dart [hunk 1]
```diff
@@ -428,50 +398,107 @@ class _VideoProgressBarState extends State<VideoProgressBar> {
...
+/// [padding] allows to specify some extra padding around the progress indicator
+/// that will also detect the gestures.
+class VideoProgressIndicator extends StatefulWidget {
+  final VideoPlayerController controller;
+  final VideoProgressColors colors;
+  final bool allowScrubbing;
+  final EdgeInsets padding;
+
+  VideoProgressIndicator(this.controller,
+      {VideoProgressColors colors,
```
Comment (mravn-google) [inline packages/video_player/lib/video_player.dart]: Trailing comma?

File: packages/video_player/lib/video_player.dart [hunk 2]
```diff
@@ -428,50 +398,107 @@ class _VideoProgressBarState extends State<VideoProgressBar> {
...
+      this.allowScrubbing,
+      this.padding: const EdgeInsets.only(top: 5.0)})
+      : colors = colors ?? new VideoProgressColors();
+
+  @override
+  _VideoProgressIndicatorState createState() {
+    return new _VideoProgressIndicatorState();
```
Comment (mravn-google) [inline packages/video_player/lib/video_player.dart]: `=>` syntax may be preferable here, depending on whether you get a line break.

File: packages/video_player/lib/video_player.dart [hunk 3]
```diff
@@ -428,50 +398,107 @@ class _VideoProgressBarState extends State<VideoProgressBar> {
...
+      final int position = controller.value.position.inMilliseconds;
+
+      int maxBuffering = 0;
+      for (DurationRange range in controller.value.buffered) {
+        final int end = range.end.inMilliseconds;
+        if (end > maxBuffering) {
+          maxBuffering = end;
+        }
+      }
+      controller.value.buffered.map((DurationRange range) => range.end);
```
Comment (mravn-google) [inline packages/video_player/lib/video_player.dart]: Remove this?

File: packages/video_player/lib/video_player.dart (hunk 2 above)
Comment (sigurdm) [reply packages/video_player/lib/video_player.dart]: done

File: packages/video_player/lib/video_player.dart (hunk 1 above)
Comment (sigurdm) [reply packages/video_player/lib/video_player.dart]: done

File: packages/video_player/lib/video_player.dart (hunk 3 above)
Comment (sigurdm) [reply packages/video_player/lib/video_player.dart]: done

**Pull request merged**